src/state/learnocaml_api.ml

```diff
@@ -101,7 +101,7 @@ type _ request =
   | Create_token:
       string * student token option * string option -> student token request
   | Create_teacher_token:
-      teacher token -> teacher token request
+      teacher token * string option -> teacher token request
```

Comment:
Dear @AltGr, thanks a lot for your very nice PR!

I didn't have the time to compile the branch and test yet (and I have an exam this afternoon at the same time of our usual slot :-/)

but FYI I've only one concern here: at first sight this change is not backward-compatible, cf.

https://github.com/ocaml-sf/learn-ocaml/blob/e060517daecc1f3c5a2fc58223f1d318bd40dd54/src/state/learnocaml_api.mli#L54-L79

Admittedly, the API `Create_teacher_token` is only called from the teacher_tab web frontend, not learn-ocaml-client, but if we we add a create-teacher-token to learn-ocaml some day, it will break compatibility.

Maybe we should discuss this

Reply:
Other idea:
don't change the API constructor `Create_teacher_token`
and do another call to the API to set the nickname, e.g., `Update_save` with a value of type:
https://github.com/ocaml-sf/learn-ocaml/blob/1584a08636eb4ddb875a2b7051f78dbc94bec9b3/src/state/learnocaml_data.ml#L70-L73

Reply:
The type changed at the code level, but the added `string option` is optional in the HTTP request: that means that the newer server version will still handle the requests from an older client in the same way as before.

At least that was the intent, I admit I haven't double-checked ; this probably wasn't included in the CI for the client.

Reply:
OK! my concern is addressed indeed, LGTM.